### PR TITLE
app-cdr/cdrtools: Fix build failure with Clang 16

### PR DESCRIPTION
**Pull request by listout**
The author of the patches is David Carlos Manuelda <StormByte@gmail.com>

Closes: https://bugs.gentoo.org/903876
Closes: https://bugs.gentoo.org/870712

### CODE REVIEW

File: app-cdr/cdrtools/files/cdrtools-3.02-fixed-autoconf-for-clang16.patch [hunk 1]
```diff
@@ -0,0 +1,1920 @@
...
++int main()
+ {
+   FILE *f=fopen("conftestval", "w");
+   if (!f) exit(1);
+@@ -1431,7 +1446,7 @@ AC_CACHE_VAL(AC_CV_NAME,
+ #ifdef	HAVE_WCHAR_H
+ #include <wchar.h>
+ #endif
+-main()
++int main()
```
Comment (orbea) [inline app-cdr/cdrtools/files/cdrtools-3.02-fixed-autoconf-for-clang16.patch:103]: Shouldn't these be `int main(void)` for `-Werror=strict-prototypes`?

File: app-cdr/cdrtools/cdrtools-3.02_alpha09-r5.ebuild [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+			elif grep -q 'StArT_T_eNdX' "${T}"/test.o ; then
+				export ac_cv_c_bitfields_htol="yes"
+			fi
+			;;
+		esac
+	fi
+}
+
+src_compile() {
+	if use unicode; then
```
Comment (juippis) [inline app-cdr/cdrtools/cdrtools-3.02_alpha09-r5.ebuild:258]: There is no such use flag in this ebuild so sourcing the ebuild fails :)

**Review by juippis — APPROVED** [review]
But the build fails to me with:
```
+ mkdir -p OBJ/amd64-linux-cc
	==> MAKING SYMLINKS in .
	==> AUTOCONFIGURING LOCAL "configure"
Autoconf requires GNU m4 1.1 or later
../RULES/local.cnf:47: OBJ/amd64-linux-cc/local.cnf: No such file or directory
make[1]: *** [../RULES/local.cnf:26: configure] Error 1
make[1]: Leaving directory '/var/tmp/portage/app-cdr/cdrtools-3.02_alpha09-r5/work/cdrtools-3.02/cdda2wav'
make: *** [RULES/rules1.dir:25: all] Error 2
 * ERROR: app-cdr/cdrtools-3.02_alpha09-r5::gentoo failed (compile phase):
 *   emake failed
```

m4 could be included as a build dep.

Although I still can't compile this with clang-16:
```
+ clang -march=native -O2 -pipe -frecord-gcc-switches -DSCHILY_BUILD -IOBJ/amd64-linux-cc -I../incs/amd64-linux-cc -I../include -I../libscg -I../libedc -I../libcdrdeflt -DCDRECORD -DDVD_MULTI -DDRV_DVD -DDRV_BD -DFIFO -DAUINFO -DHAVE_LIB_EDC_ECC -DCLONE_WRITE -DVENDOR_CMDS -DUSE_LARGEFILES -DSCHILY_PRINT '-DINS_BASE="/var/tmp/portage/app-cdr/cdrtools-3.02_alpha09-r5/image/usr"' '-DTEXT_DOMAIN="SCHILY_cdrtools"' -D_GNU_SOURCE -c -o OBJ/amd64-linux-cc/cdrecord.o cdrecord.c
cdrecord.c:280:1: warning: a function definition without a prototype is deprecated in
      all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
main(ac, av)
^
cdrecord.c:508:7: error: call to undeclared function 'mlockall'; ISO C99 and later do
      not support implicit function declarations [-Wimplicit-function-declaration]
                if (mlockall(MCL_CURRENT|MCL_FUTURE) < 0) {
```

**Comment by billie80** [comment]
Both referenced bugs have been fixed in a different way.

https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=d479020a18c69c066f3ec1e8862e30d3c84422ee
https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=c79fce973fb6d30324caf13ed1ef7881fb9741c9

**Pull request closed without merging**